### ISSUE TITLE
PayPal Express Checkout popup close immediately | react-paypal-express-checkout

### ISSUE THREAD

**Issue by thelalitmohan** [thinhvo0108/react-paypal-express-checkout#52]
Getting the AUTHENTICATION_FAILURE error to the following API
https://www.sandbox.paypal.com/v1/payment-experience/web-profiles

`{"name":"AUTHENTICATION_FAILURE","debug_id":"4ce6b62baa64","message":"Authentication failed due to invalid authentication credentials or a missing Authorization header","information_link":"https://developer.paypal.com/docs/api/payment-experience/#errors","details":[]}`

Can you please review and fix it

**Comment by konapaz** [comment]
I have the same issue only in the sandbox mode

{"name":"AUTHENTICATION_FAILURE","debug_id":"7105b07666045","message":"Authentication failed due to invalid authentication credentials or a missing Authorization header","information_link":"https://developer.paypal.com/docs/api/payment-experience/#errors","details":[]}

Did paypal change something for sandbox on April or June 2018 ?

Similar issues...

https://github.com/paypal/paypal-checkout-components/issues/356

https://www.paypal-community.com/t5/REST-APIs/Authentocation-error-for-client-rest-API-in-sandbox/td-p/1495921

**Comment by thelalitmohan** [comment]
@konapaz, the sandbox was fine 10 days ago. production is still good.

**Comment by konapaz** [comment]
> @konapaz, the sandbox was fine 10 days ago. production is still good.

@lalitmohan001 So, Paypal something changed in its API

**Comment by nathanmacfarlane** [comment]
I'm having the same issue. Any solutions?

**Comment by thelalitmohan** [comment]
@nathanmacfarlane, yes PayPal made some changes into their API. I see, there is no commit to the repository from the last 1 year.

**Comment by lukey78** [comment]
Let's find out what changes have been made, fix it through a pull request, and hope that the change in production will not happen tomorrow. Any ideas where to start?

It seems the Express Button uses this thing:

https://developer.paypal.com/docs/classic/express-checkout/in-context/enable-in-context-checkout/

And that one is deprecated since 2017. 

Also, the REST API v1 which is referenced in the docs is also deprecated.
I didn't find out yet how long this stuff will be supported on production.

**Comment by lukey78** [comment]
This seems to be the current "easy" Javascript implementation:
https://developer.paypal.com/docs/checkout/

**Comment by thelalitmohan** [comment]
@lukey78, better to user javascript implementation. Package production can be down anytime like a sandbox. Please ignore this package

**Comment by aralroca** [comment]
Same issue... Any new?

**Comment by aralroca** [comment]
I found why is not working...

```js
export default scriptLoader('https://www.paypalobjects.com/api/checkout.js')(PaypalButton);
```
In https://github.com/thinhvo0108/react-paypal-express-checkout/blob/master/src/index.js#L109

And then in the PayPal docs...

https://developer.paypal.com/docs/archive/checkout/how-to/customize-button/

![image](https://user-images.githubusercontent.com/13313058/70132258-232f0680-1684-11ea-8075-5945e3f445b7.png)

**Comment by aralroca** [comment]
An alternative... https://www.npmjs.com/package/react-paypal-button-v2

**Comment by masoud46** [comment]
Same issue!

**Comment by seifeslimene** [comment]
Same issue!
